internal/docstore/mongodocstore/mongo.go

```diff
@@ -649,22 +652,31 @@ func (c *collection) unorderedBulkWrite(ctx context.Context, iactions []indexedA
 			if newID != nil {
 				newIDs[i] = newID
 			}
+			if err == nil {
+				nCreate++
+			}
 		case driver.Delete:
 			m, err = c.newDeleteModel(a)
 			if err == nil {
 				nDeletes++
 			}
-		case driver.Replace, driver.Put:
-			m, err = c.newReplaceModel(a, a.Kind == driver.Put)
-			if err != nil && a.Kind != driver.Put {
-				nMatches++
+		case driver.Replace:
+			m, err = c.newReplaceModel(a, false)
+			if err == nil {
+				nNonCreateWrite++
+				nModified++
+			}
+		case driver.Put:
+			m, err = c.newReplaceModel(a, true)
+			if err == nil {
+				nNonCreateWrite++
```

Comment:
Are you sure that a Put to an existing document doesn't show up as a modification? (Is there a test for that?)